test/tests/functional/pbs_cray_suspend_resume.py

```diff
@@ -93,3 +97,390 @@ def test_exclusive_job_not_suspended(self):
+                ATTR_l + '.walltime': '00:01:00',
+                ATTR_l + '.place': 'scatter',
+                ATTR_q: 'highp'})
+        j2id = self.server.submit(j2)
+
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=j2id)
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid)
+
+        self.server.delete(j2id)
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        self.server.expect(SERVER, {'server_state': 'Active'})
+        self.server.expect(JOB, {ATTR_state: 'S'}, id=jid)
+        self.server.cleanup_jobs()
+
+    def test_multi_express(self):
+        """
+        Test of multiple express queues of different priorities.
+        See that jobs from the higher express queues preempt jobs
+        from lower express queues.  Also see when express jobs finish
+        (or are deleted), suspended jobs restart.
+        Make sure loadLimit is set to 4 on the server node:
+        # apmgr config loadLimit 4
+        """
+
+        _t = ('\"express_queue, normal_jobs, server_softlimits,' +
+              ' queue_softlimits\"')
+        a = {'preempt_prio': _t}
+        self.scheduler.set_sched_config(a)
+
+        try:
+            self.server.manager(MGR_CMD_DELETE, QUEUE, id="expressq")
+        except:
+            pass
+        a = {'queue_type': 'e',
+             'started': 'True',
+             'enabled': 'True',
+             'Priority': 150}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "expressq")
+
+        try:
+            self.server.manager(MGR_CMD_DELETE, QUEUE, id="expressq2")
+        except:
+            pass
+        a = {'queue_type': 'e',
+             'started': 'True',
+             'enabled': 'True',
+             'Priority': 160}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "expressq2")
+
+        try:
+            self.server.manager(MGR_CMD_DELETE, QUEUE, id="expressq3")
+        except:
+            pass
+        a = {'queue_type': 'e',
+             'started': 'True',
+             'enabled': 'True',
+             'Priority': 170}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, "expressq3")
+
+        # Count the compute nodes
+        vlist = []
+        try:
+            vnl = self.server.filter(
+                VNODE, {'resources_available.vntype': 'cray_compute'})
+            vlist = vnl["resources_available.vntype=cray_compute"]
+        except Exception:
+            sys.exc_clear()
+            self.logger.info("There are no cray_compute vnodes present.")
+        numvnl = len(vlist)
+
+        nodes = self.server.status(NODE)
+        for n in nodes:
+            if 'resources_available.vntype' in n:
+                if n['resources_available.vntype'] == 'cray_compute':
+                    cpus = n['resources_available.ncpus']
+                    ncpus = int(cpus)
+
+        j1 = Job(TEST_USER, {ATTR_l + '.select': '%d:ncpus=%d' % (numvnl, ncpus),
```

Comment:
If the number of cpus are not the same across the compute nodes (for example node1 has 8 cpus, but node2 has 24), won't this job fail to run?  Can we just request a few ncpus per chunk and use place=scatter to cause the job to run across all the nodes?